### PR TITLE
pkg/asset: Update kube-dns to v17.1

### PR DESCRIPTION
**Pull request by dghubble**
Update kube-dns deployment from 11 to v17.1.

rel: https://github.com/coreos/coreos-kubernetes/commit/64e44562fd4da459d631d28f6269c22779372cca
replaces: #107
cc: @ericchiang

### CODE REVIEW

**Comment by aaronlevy** [comment]
lgtm

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -369,23 +349,21 @@ spec:
...
             cpu: 10m
-            memory: 20Mi
+            memory: 50Mi
           requests:
             cpu: 10m
-            memory: 20Mi
+            memory: 50Mi
         args:
-        - -cmd=nslookup kubernetes.default.svc.cluster.local 127.0.0.1 >/dev/null
+        - -cmd=nslookup kubernetes.default.svc.cluster.local 127.0.0.1 >/dev/null && nslookup kubernetes.default.svc.cluster.local 127.0.0.1:10053 >/dev/null
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Why the additional check? This doesn't seem to be in the upstream version

Comment (dghubble) [inline pkg/asset/internal/templates.go]: Mirrored the update done in coreos-kubernetes https://github.com/coreos/coreos-kubernetes/blob/master/multi-node/generic/controller-install.sh#L557

**Comment by dghubble** [comment]
I'll verify this through coreos-baremetal soon.

File: pkg/asset/internal/templates.go (hunk 1 above)
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: huh, odd. I'll have to go back and see where that came from.

For consistency with upstream, can you update this to more closely reflect: https://github.com/kubernetes/kubernetes/blob/release-1.3/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base

(but s/replicationController/Deployment)

Comment (dghubble) [inline pkg/asset/internal/templates.go]: Are you wanting the YAML comments as well?

Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: I don't have strong feelings - but if they add useful context may as well leave them in (also makes the copy/paste job a little easier).

**Comment by aaronlevy** [comment]
lgtm

**Comment by dghubble** [comment]
Found a typo while validating.

**Pull request merged**